### PR TITLE
New alias TfsPullRequestLastSourceCommit 

### LINKED ISSUE #8: Possibility to determine the hash of the last commit on the source branch
Possibility to determine the hash of the last commit on the source branch

### PR DESCRIPTION
**Pull request by stffabi**
Makes it possible to retrieve the latest commit of the source branch of the pull request. Fixes #8

### CODE REVIEW

**Comment by stffabi** [comment]
@pascalberger can you please take a look into the build infrastructure? It seems like we need some updating of the cake infrastructure to be able to build.

**Comment by gep13** [comment]
@stffabi @pascalberger you may need to add a cake.config file with these contents:

https://github.com/cake-build/resources/blob/develop/cake.config

In order for the build to work.

**Comment by gep13** [comment]
@stffabi ah, yes, the latest version of Cake.Recipe assumes that you are using the latest version of Cake, so that change is required.

**Comment by gep13** [comment]
@stffabi the current error message is due to the fact that the NuGet package for the ReSharper Command Lines has a HUGE name, which causes problems when running on Windows.  You can fix that by adding this line:

https://github.com/GitTools/GitReleaseManager/blob/develop/cake.config#L6

To your cake.config file.

**Comment by gep13** [comment]
This will make Cake use an In Process version of NuGet, which isn't subject to the same problem, as it doesn't create the intermediate folder in the tools folder, which cases it to go over the allowed length.

**Comment by stffabi** [comment]
Ah thanks @gep13, will try that.

**Comment by gep13** [comment]
@stffabi ok, the final build error that you have there I don't have an answer for 😢 

You can skip that though by adding this:

https://github.com/cake-contrib/Cake.SmartAssembly/blob/master/setup.cake#L15

I need to look at what has changed with CodeCov in this area.

**Comment by stffabi** [comment]
@gep13 thanks for your support.

**Comment by gep13** [comment]
@stffabi you are very welcome!  Would love to get to the bottom of this problem, hence why I was watching your build to see if it was going to suffer the same fate 😢

File: cake.config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[Nuget]
+; Use the new InProcess client, because same packages have long file paths.
```
Comment (pascalberger) [inline cake.config]: Typo: `because some packages`

**Comment by gep13** [comment]
Woot! Got there in the end 😄

File: cake.config (hunk 1 above)
Comment (stffabi) [reply cake.config]: fixed

**Pull request merged**